src/c++/perf_analyzer/genai-perf/genai_perf/llm_metrics.py

```diff
@@ -600,13 +600,13 @@ def _preprocess_response(
 
             # Remove responses without any content
             # These are only observed to happen at the start or end
-            while res_outputs[0] and self._is_openai_empty_response(
+            while res_outputs and self._is_openai_empty_response(
```

Comment:
Do we need additional errors/warnings if all responses are null, though?

Reply:
wont PA error before that?
with "no responses in window"?

Reply:
I think this is different than that. This is still a response, it's just an "empty" response.

Reply:
Sometimes models output only a empty string `""` and this is what's causing this. Since this is a known behavior of LLM, and this doesn't affect metric numbers -- as long as users send enough requests (to get good statistics) -- I don't think warning is necessary.

Reply:
Good catch, Matt! It affects metric numbers because the models are potentially outputting unexpected outputs, right? And this may make metrics look better/worse than they are, even if the job of PA/GenAI-Perf is not explicitly to perform validation.

Is the empty string case a known correct case? It feels like if models are returning empty strings, something is most likely going wrong. Have we investigated why they are sometimes outputting only an empty string? If that is a desirable case under certain conditions, I feel like we should be allowing a flag that lets users accept empty strings, not accept them by default.

Reply:
If there are no responses left (empty list), metric calculation is skipped and therefore there's no affect to the metrics.

Reply:
I don't think empty output is an unexpected output from the model AFAIK. And that's why we have this while loop to remove all empty response outputs.

Reply:
@IzzyPutterman

Reply:
Do you think it is still worth it to include this response as a TTFT?  But probably best to skip, I am not sure how to work around it for req/s

Reply:
I think the decision was to ignore all the first responses with empty content since TTFT of empty response doesn't really makes sense.

Reply:
Ahh, I am confusing the empty content case with the first response being DONE.  If the first response is DONE, would we want that to count towards TTFT?

Reply:
Just to clarify,
- Empty response: response with `"content": ""` (empty output string).
- Null response: either response with missing `"content"` field or response is `data: [DONE]`.

Given that, is it normal (or even possible) to have `data: [DONE]` as first response? And I don't think it makes sense to include that to our TTFT metric since that is not a normal response from the model.

Reply:
@IzzyPutterman Can model output empty string (e.g. `""`) as output?

Reply:
I think regardless of whether a server/model outputting all "empty" responses is a bug in the server/model, this fix improves the usability of GenAI-Perf by not crashing and still allowing largely valid output to be output to user.

But we should try to confirm if all-empty responses is a bug or not in the server/model. If we think it's unexpected behavior, we could file an issue with the server implementers to start (maybe it's not from them, maybe it's more relevant for the model developers?)